packages/react/src/components/Checkbox/index.tsx

```diff
@@ -77,8 +79,18 @@ const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
             type="checkbox"
             checked={isChecked}
             disabled={disabled}
-            onFocus={(): void => setFocused(true)}
-            onBlur={(): void => setFocused(false)}
+            onFocus={(e): void => {
+              setFocused(true);
+              if (onFocus) {
```

Comment:
We typically do `typeof` checks for safeguarding:

```suggestion
              if (typeof onFocus === 'function') {
```